components/backup_text/src/mask.rs

```diff
@@ -109,77 +113,79 @@ fn mask_time(ctx: &mut EvalContext, time: DateTime) -> DateTime {
     let second = unchecked_time.second() % 60; // 0..59
     let micro = unchecked_time.micro() % 1000000; // 0..999999
 
-    DateTime::from_slice(
+    let time = DateTime::from_slice(
         ctx,
         &[year, month, day, hour, minute, second, micro][..],
         time_type,
         fsp,
     )
-    .unwrap()
+    .ok_or_else(|| anyhow!("failed to mask time"))?;
+
+    Ok(time)
 }
 
-fn format_float(f: f64, neg: bool, mut int_num: usize, frac: usize) -> String {
-    if int_num < 1 {
-        int_num = 1;
+fn format_float(mut f: f64, neg: bool, mut int_num: usize, frac: usize) -> String {
+    f = f.abs();
+    int_num = int_num.max(1);
+
+    let mut s: String = format!("{:.10e}", f)
+        .chars()
+        .filter(|c| c.is_numeric())
+        .collect();
+
+    if s.len() < int_num {
```

Comment:
TiDB side should also adopt this change:
https://github.com/BugenZhao/sql-masker/blob/d953e85be293f97476ef6bbd65e75744d223a353/mask/funcs/workload_sim.go#L160